File: data/python/CHANGE_ARRAY_PERMUTATION_NUMBERS_1_N_TRANSLATED.java
====================
public static void f_gold ( int [ ] a , int n ) {
  Map < Integer , Integer > count = Maps . newHashMap ( ) ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( count . containsKey ( a [ i ] ) ) {
      count . put ( a [ i ] , 1 ) ;
    }
    else {
      count . put ( a [ i ] , 1 ) ;
      ;
    }
  }
  int nextMissing = 1 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( count . get ( a [ i ] ) != 1 || a [ i ] > n || a [ i ] < 1 ) {
      count . put ( a [ i ] , 1 ) ;
      while ( count . containsKey ( nextMissing ) ) {
        nextMissing ++ ;
      }
      a [ i ] = nextMissing ;
      count . put ( nextMissing , 1 ) ;
    }
  }
}

File: data/java/CHANGE_ARRAY_PERMUTATION_NUMBERS_1_N_TRANSLATED.py
====================
def f_gold ( a , n ) :
    count = { }
    for i in range ( n ) :
        if count.has_key ( a [ i ] ) :
            count [ a [ i ] ] = count [ a [ i ] ] + 1
        else :
            count [ a [ i ] ] = 1
    next_missing = 1
    for i in range ( n ) :
        if count.has_key ( a [ i ] ) and count [ a [ i ] ] != 1 or a [ i ] > n or a [ i ] < 1 :
            count [ a [ i ] ] = count [ a [ i ] ] - 1
            while count.has_key ( next_missing ) :
                next_missing += 1
            a [ i ] = next_missing
            count [ next_missing ] = 1
    return count

